Update environment variable format to be standard on all (#2029)
nautobot/nautobot: nautobot/docs/configuration/optional-settings.md

* [RELEASE_CHECK_TIMEOUT](#release_check_timeout)
* [RELEASE_CHECK_URL](#release_check_url)

## Nautobot Config Settings

> The below Nautobot Config settings, generally managed within `nautobot_config.py` or Environment Variables provide the ability to configure your application.

Reviewer: ```suggestion
The configuration options in this section can be defined within `nautobot_config.py` to configure your Nautobot application. Many can also alternatively be defined by environment variables, as noted under each option where applicable.
```

Default: `None`

> Environment Variable: `SCRIPT_NAME`

Author: Please verify this one, I believe it is correct, but work another look.

Reviewer: I'm not an expert on this feature but I don't think this is correct. If I'm reading the Django docs correctly, the idea is that if `$SCRIPT_NAME` is set in your environment / on your HTTP server, you must *also* set `settings.FORCE_SCRIPT_NAME` to match that value.

Author: Honestly, I still don't get it :) 

The way it is worded originally, I am not sure what is being requested. Would you mind making a suggestion to what you think is correct?

Reviewer 2: Yeah this is a bit confusing. This is a server internal environment variable

Default: `os.path.join(NAUTOBOT_ROOT, "git")`

Environment Variable: `NAUTOBOT_GIT_ROOT`

The file path to a directory where cloned [Git repositories](../models/extras/gitrepository.md) will be located.

---

## GIT_SSL_NO_VERIFY

Default: Unset

Environment Variable: `GIT_SSL_NO_VERIFY`

If you are using a self-signed git repository, you will need to set the environment variable `GIT_SSL_NO_VERIFY="1"`

> in order for the repository to sync.

Reviewer: This one's a bit confusing -- *only* the environment variable `$GIT_SSL_NO_VERIFY` is relevant. Configuring `settings.GIT_SSL_NO_VERIFY` would do nothing. I feel like adding a separate `Environment Variable:` entry here is actually a bit more confusing versus just reading the textual description, personally.

Default: `~/.nautobot/`

Environment Variable: `NAUTOBOT_ROOT`

> 

Reviewer: LIkewise here, environment variable `$NAUTOBOT_ROOT` is the only relevant setting - `settings.NAUTOBOT_ROOT` does nothing. 

Maybe instead of adding `Environment Variable:` entries to GIT_SSL_NO_VERIFY and NAUTOBOT_ROOT, we should move these environment-variable-only settings to a new, separate section of this document?

* `NAUTOBOT_SHORT_DATETIME_FORMAT`

---

## Environment-Variable-Only Settings

!!! warning
    The following settings are **only** configurable as environment variables, and not via `nautobot_config.py` or similar.

---

### GIT_SSL_NO_VERIFY

Default: Unset

If you are using a self-signed git repository, you will need to set the environment variable `GIT_SSL_NO_VERIFY="1"`
in order for the repository to sync.

!!! warning
    This _must_ be specified as an environment variable. Setting it in `nautobot_config.py` will not have the desired effect.

---

### NAUTOBOT_ROOT

Default: `~/.nautobot/`

The filesystem path to use to store Nautobot files (Jobs, uploaded images, Git repositories, etc.).

This setting is used internally in the core settings to provide default locations for [features that require file storage](../../configuration/#file-storage), and the [default location of the `nautobot_config.py`](../../configuration/#specifying-your-configuration).

!!! warning
    Do not override `NAUTOBOT_ROOT` in your `nautobot_config.py`. It will not work as expected. If you need to customize this setting, please always set the `NAUTOBOT_ROOT` environment variable.

> ## WSGI Config Settings

Author: ```suggestion
## Django Settings
```

The below setting is documented for WSGI for clarity purposes.

### FORCE_SCRIPT_NAME

Default: `None`

If not `None`, this will be used as the value of the `SCRIPT_NAME` environment variable in any HTTP request. This setting can be used to override the server-provided value of `SCRIPT_NAME`, which is most commonly used for hosting Nautobot in a subdirectory (e.g. _example.com/nautobot/_).

!!! important
    To host Nautobot under a subdirectory you must set this value to match the same prefix configured on your HTTP server. For example, if you configure NGINX to serve Nautobot at `/nautobot/`, you must set `FORCE_SCRIPT_NAME = "/nautobot/"`.

> Please see the [official Django documentation on `FORCE_SCRIPT_NAME`](https://docs.djangoproject.com/en/stable/ref/settings/#force-script-name) for more information.

Reviewer: So why did this need to be moved to a separate section of its own, if it's a Django setting that you specify in `nautobot_config.py`?

Author: I am not tracking your question? My understanding is there are Nautobot Config settings that are here in one group and Django settings in another group, I would have that we should separate. 

Anything else you are hinting at, is not resonating with me now.

Reviewer: My question is what's special/different about this setting that it needs to be separate from all the other configs? I'm not seeing why `FORCE_SCRIPT_NAME` (a Django setting that is configured in `nautobot_config.py`) is different than `ADMINS`, `DEBUG`, `HTTP_PROXIES`, `LOGGING`, etc. (also Django settings that are configured in `nautobot_config.py`). Why not leave it where it was?

Author: I think you are highlighting the confusion I am having with the documentation. It seems that we have a mix of Django settings (ADMINS, DEBUG, HTTP_PROXIES, LOGGING, etc), with Nautobot only settings (BRANDING_FILEPATHS, NAPALM_USERNAME, etc). Why are we documenting both? Why not document `CSRF_USE_SESSIONS` (which is a valid django setting) as an example.

There may be very good reasons to document django settings here, rather than refer to Django settings, but I do not understand what the decision tree is right now. I think either of the following are good steps forward:
* Split all of the Django settings from Nautobot settings
* Document why some Django settings are shown and others are not (e.g. the ones shown are most likely to cause issues)

Reviewer: Fair points, but perhaps out of scope for the original stated purpose of this PR? :-) Can we scope this down to just addressing the environment-variables-as-distinct-from-config-variables issue, and get a follow up documentation/housekeeping issue open to revisit whether we should "undocument" some of the Django built-in configs variables?

Author: Updated created #2153 to track.

* `NAUTOBOT_SHORT_TIME_FORMAT`
* `NAUTOBOT_DATETIME_FORMAT`
* `NAUTOBOT_SHORT_DATETIME_FORMAT`

> 

Reviewer: ```suggestion

---

```

> ## Settings as Only Environment Variables

Reviewer: Perhaps?

```suggestion
## Environment-Variable-Only Settings
```

> The following settings are **only** configurable as Environment Variables, and not via `nautobot_config.py` or similar.

Reviewer: Perhaps?

```suggestion
!!! warning
    The following settings are **only** configurable as environment variables, and not via `nautobot_config.py` or similar.
```

---

### GIT_SSL_NO_VERIFY

Default: Unset

> 
> Environment Variable: `GIT_SSL_NO_VERIFY`

Reviewer: This is redundant now.

```suggestion
```

If you are using a self-signed git repository, you will need to set the environment variable `GIT_SSL_NO_VERIFY="1"`
in order for the repository to sync.

> 
> !!! warning
>     This _must_ be specified as an environment variable. Setting it in `nautobot_config.py` will not have the desired effect.

Reviewer: I think this is also not needed after the reorganization.
```suggestion
```

Author: I would prefer to keep these 3 open ones in. While it is clear from the headings that each section is different, as the reader it is easy to lose that context when reading that specfic section and the lions share is training you to think differently. Not going to lose sleep, so feel free to commit suggested or resolve.

---

### NAUTOBOT_ROOT

Default: `~/.nautobot/`

> 
> Environment Variable: `NAUTOBOT_ROOT`

Reviewer: ```suggestion
```

The filesystem path to use to store Nautobot files (Jobs, uploaded images, Git repositories, etc.).

This setting is used internally in the core settings to provide default locations for [features that require file storage](../../configuration/#file-storage), and the [default location of the `nautobot_config.py`](../../configuration/#specifying-your-configuration).

> 
> !!! warning
>     Do not override `NAUTOBOT_ROOT` in your `nautobot_config.py`. It will not work as expected. If you need to customize this setting, please always set the `NAUTOBOT_ROOT` environment variable.

Reviewer: ```suggestion
```

The function must take only one argument: the device display name, as a string, attempting to be rendered on the rack elevation.

The function must return only one argument: a string of the truncated device display name.

## Environment-Variable-Only Settings

!!! warning
    The following settings are **only** configurable as environment variables, and not via `nautobot_config.py` or similar.

---

### GIT_SSL_NO_VERIFY

Default: Unset

If you are using a self-signed git repository, you will need to set the environment variable `GIT_SSL_NO_VERIFY="1"`
in order for the repository to sync.

!!! warning
    This _must_ be specified as an environment variable. Setting it in `nautobot_config.py` will not have the desired effect.

---

### NAUTOBOT_ROOT

Default: `~/.nautobot/`

The filesystem path to use to store Nautobot files (Jobs, uploaded images, Git repositories, etc.).

This setting is used internally in the core settings to provide default locations for [features that require file storage](../../configuration/#file-storage), and the [default location of the `nautobot_config.py`](../../configuration/#specifying-your-configuration).

!!! warning

>     Do not override `NAUTOBOT_ROOT` in your `nautobot_config.py`. It will not work as expected. If you need to customize this setting, please always set the `NAUTOBOT_ROOT` environment variable.

Author: ```suggestion
    Do not override `NAUTOBOT_ROOT` in your `nautobot_config.py`. It will not work as expected. If you need to customize this setting, please always set the `NAUTOBOT_ROOT` environment variable.

```
